serve.go

```diff
@@ -1,10 +1,112 @@
 package main
 
 import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
 	"log"
 	"net/http"
+	"os"
+	"os/signal"
+	"path/filepath"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+	"unicode"
+	"unicode/utf8"
+
+	"github.com/gomarkdown/markdown/ast"
+	"github.com/gomarkdown/markdown/parser"
+)
+
+const (
+	baseTimeout            = 10 * time.Second
+	mdExt           string = `.md`
+	maxMarkdownSize        = 0x800000 //8MB which is bonkers
+)
+
+var (
+	lp     = flag.Uint("listen-port", 3001, "listening port")
+	la     = flag.String("listen-address", "", "listening address")
+	sdir   = flag.String("host-dir", ".", "Root directory for files")
+	blink  = flag.String("base-link", "/#!", "Base value to return for all links")
+	fdebug = flag.Bool("debug", false, "Output a ton of info while running")
 )
 
+type server struct {
+	sync.Mutex
+	http.Server
+	blink string //the base link anchor
+	dir   string
+	links map[string][]string
+}
+
+func main() {
+	flag.Parse()
+	if *lp == 0 || *lp > 0xffff {
```

Comment:
This is somewhat unnecessary  as ListenAndServe() will fail when it attempts to bind.

Reply:
i know, but i like throwing a super specific error.